### ISSUE TITLE
Roofline Model and HLO Op Profile are empty for JAX GPU traces when XLA command buffers are enabled

### ISSUE THREAD

**Issue by Florentino-Ding** [openxla/xprof#3070]
## Summary

When profiling a JAX workload on an NVIDIA GPU with the default XLA command-buffer configuration, XProf successfully captures GPU activity and displays kernel statistics, but the **HLO Op Profile**, **HLO Op Stats**, and **Roofline Model** contain no useful operation-level data.

Running the same workload in a fresh process with GPU command buffers disabled fixes all three tools:

```bash
XLA_FLAGS='--xla_gpu_enable_command_buffer=' \
  python token_mixer_nnx.py
```

I would like to confirm whether losing per-HLO attribution for kernels executed through an XLA command buffer is an expected limitation or an XProf bug.

## Environment

- Python: 3.14.6
- JAX: 0.11.0
- jaxlib: 0.11.0
- jax-cuda12-plugin: 0.11.0
- XProf: 2.22.3
- Accelerator: NVIDIA Ampere GPU
- Profile collected with `jax.profiler.trace`
- Workload is compiled and warmed up before profile collection

## Reproduction

The relevant profiling pattern is:

```python
# Compile and warm up outside the trace.
for _ in range(10):
    forward(params, inputs).block_until_ready()

with jax.profiler.trace(
    "profiling_results/token_mixer",
    create_perfetto_link=False,
):
    for _ in range(50):
        output = forward(params, inputs)
        output.block_until_ready()
```

### Case 1: Default XLA configuration

```bash
python token_mixer_nnx.py
xprof profiling_results/token_mixer
```

Observed results:

- Trace Viewer contains GPU streams and kernel executions.
- Kernel Stats is populated.
- Approximately 2,600 GPU kernel events are present.
- Every kernel is attributed to the same synthetic HLO operation:
  `command_buffer_1`.
- HLO Op Stats has zero rows.
- HLO Op Profile collapses to an `IDLE` root with no children.
- Roofline Model contains only zero-valued program totals, with no operation-level points.
- The UI does not explain that command-buffer attribution is preventing these tools from working.

A representative GPU event contains metadata similar to:

```text
hlo_module: jit_token_mixer
hlo_op: command_buffer_1
cuda_graph_id: 2
tf_op: XlaModule:
```

### Case 2: GPU command buffers disabled

Run the exact same script in a fresh process:

```bash
XLA_FLAGS='--xla_gpu_enable_command_buffer=' \
  python token_mixer_nnx.py
```

Observed results:

- Kernels are attributed to individual HLO operations.
- 50 distinct HLO operations are visible.
- HLO Op Stats contains 50 rows.
- HLO Op Profile contains a populated program tree.
- Roofline Model contains both program-level and operation-level data
  (102 rows in my test).

No workload changes are required; only the XLA flag differs.

I also tried:

```bash
XLA_FLAGS='--xla_enable_command_buffers_during_profiling=false'
```

This did not restore HLO attribution when the executable had already been compiled and warmed up before entering `jax.profiler.trace`.

## Expected behavior

Ideally, one of the following would happen:

1. XProf correlates kernels inside an XLA command buffer with their original HLO operations, allowing HLO Op Profile and Roofline Model to work with the default XLA configuration; or
2. If this correlation is not currently supported, XProf clearly reports that these tools are unavailable because the captured kernels are attributed only to a command buffer, and documents the required workaround.

A completely empty HLO Op Profile and Roofline Model currently looks like a failed or invalid profile even though kernel collection succeeded.

## Relationship to [#2544](https://github.com/openxla/xprof/issues/2544)

This appears closely related to
[#2544](https://github.com/openxla/xprof/issues/2544), where an individual
kernel’s HLO graph cannot be viewed because its XLA operation is represented
only as `command_buffer_x`.

The behavior reported here appears to be the same underlying loss of
per-HLO attribution, but it also affects aggregate analysis tools:

- HLO Op Stats
- HLO Op Profile
- Roofline Model

## Questions

1. Is loss of per-HLO attribution inside GPU command buffers expected?
2. Is disabling command buffers the recommended way to collect an
   operation-level XProf profile?
3. Is there a supported way to retain command-buffer execution while still
   recording the original HLO operation for each kernel?
4. If this is currently unsupported, could XProf display a diagnostic instead
   of showing empty analysis pages?

I can attach both `.xplane.pb` files—the default capture and the
command-buffer-disabled capture—if they would help with debugging.

**Comment by Florentino-Ding** [comment]
Hi! I’d like to provide a small update after some additional investigation.

From what I’ve observed so far:
- When '--xla_gpu_enable_command_buffer=' flag is added, everything work fine.
- However, when TensorCore operations are evolved (through cuBLAS), the problem happens again and currently no solution was found.
- Nsight System successfully captured useful kernel information.

Would anyone familiar with XProf tracing be willing to take a look when you have a chance?

@zzzaries @Matt-Hurd @cliveverghese 
Sorry for the ping, but I noticed you’ve recently been active in XProf-related discussions, so I thought you might have some insight. Any suggestions or pointers would be greatly appreciated. Thanks!
